Userland/Utilities/strings.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/CharacterTypes.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/Stream.h>
+#include <LibCore/System.h>
+#include <LibMain/Main.h>
+#include <unistd.h>
+
+enum class StringOffsetFormat {
+    None = 0,
+    Decimal,
+    Octal,
+    Hexadecimal
+};
+
+static bool should_print_characters(Vector<u8>& characters)
+{
+    for (u8 ch : characters) {
+        if (is_ascii_printable(ch) && !is_ascii_space(ch))
+            return true;
+    }
+    return false;
+}
+
+static ErrorOr<void> print_characters(Vector<u8>& characters, StringOffsetFormat string_offset_format, size_t string_offset_position)
+{
+    switch (string_offset_format) {
+    case StringOffsetFormat::Decimal:
+        outln("offset {:d}: ", string_offset_position);
+        break;
+    case StringOffsetFormat::Octal:
+        outln("offset 0o{:o}: ", string_offset_position);
+        break;
+    case StringOffsetFormat::Hexadecimal:
+        outln("offset 0x{:x}: ", string_offset_position);
+        break;
+    default:
+        break;
+    }
+    TRY(Core::System::write(STDOUT_FILENO, characters.span()));
```

Comment:
Doesn't this work?
```cpp
outln("{}", characters.span());
```

Reply:
Needs formatting -  `outln("{:s}", characters.span());` works great :^)